CodingChallenges/CC_051_astar/Processing/CC_051_astar/CC_051_astar.pde

```diff
@@ -0,0 +1,169 @@
+// Daniel Shiffman
+// http://codingtra.in
+// http://patreon.com/codingtrain
+// Part 1: https://youtu.be/aKYlikFAV4k
+// Part 2: https://youtu.be/EaZxUCWAjb0
+// Part 3: https://youtu.be/jwRT4PCT6RU
+// Processing transcription: Chuck England
+
+// An educated guess of how far it is between two points
+float heuristic(Spot a, Spot b) {
+  float d = dist(a.i, a.j, b.i, b.j);
+  // float d = abs(a.i - b.i) + abs(a.j - b.j);
+  return d;
+}
+
+// How many columns and rows?
+int cols = 50;
+int rows = 50;
+
+// This will be the 2D array
+Spot[][] grid = new Spot[cols][rows];
+
+// Open and closed set
+List<Spot> openSet = new ArrayList<Spot>();
+List<Spot> closedSet = new ArrayList<Spot>();
+
+// Start and end
+Spot start;
+Spot end;
+Spot current;
+
+// Width and height of each cell of grid
+float w, h;
+
+// The road taken
+List<Spot> path = new ArrayList<Spot>();
+
+void setup() {
+  size(400, 400);
+  println("A*");
+
+  // Grid cell size
+  w = width / (float)cols;
```

Comment:
```suggestion
  w = float(width) / cols;
```

Processing has a `float` method which is admittedly a little weird, but makes the code a bit easier on the eyes for beginners.